src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -702,89 +719,122 @@ private static GapicProductConfig createDummyInstance(
-            configProto.getCollectionOneofsList(),
-            singleResourceNameConfigsFromGapicConfig,
-            fixedResourceNameConfigs,
-            file);
-    if (diagCollector.getErrorCount() > 0) {
-      ToolUtil.reportDiags(diagCollector, true);
-      return null;
+            resource,
+            protoFile,
+            protoParser,
+            fullyQualifiedSingleResourcesFromProtoFileBuilder);
+      }
     }
 
     ImmutableMap<String, SingleResourceNameConfig>
         fullyQualifiedSingleResourceNameConfigsFromProtoFile =
-            createSingleResourceNameConfigsFromProtoFile(diagCollector, resourceDefs, protoParser);
+            ImmutableMap.copyOf(fullyQualifiedSingleResourcesFromProtoFileBuilder);
+    ImmutableMap<String, FixedResourceNameConfig>
+        fullyQualifiedFixedResourceNameConfigsFromProtoFile =
+            ImmutableMap.copyOf(fullyQualifiedFixedResourcesFromProtoFileBuilder);
+
     ImmutableMap<String, ResourceNameOneofConfig> resourceNameOneofConfigsFromProtoFile =
         createResourceNameOneofConfigsFromProtoFile(
             diagCollector,
             fullyQualifiedSingleResourceNameConfigsFromProtoFile,
-            fixedResourceNameConfigs,
+            fullyQualifiedFixedResourceNameConfigsFromProtoFile,
             resourceSetDefs,
             protoParser);
 
     // Populate a SingleResourceNameConfigs map, using just the unqualified names.
-    Map<String, SingleResourceNameConfig> singleResourceConfigsFromProtoFile = new HashMap<>();
+    LinkedHashMap<String, SingleResourceNameConfig> singleResourceConfigsFromProtoFile =
+        new LinkedHashMap<>();
     for (String fullName : fullyQualifiedSingleResourceNameConfigsFromProtoFile.keySet()) {
       int periodIndex = fullName.lastIndexOf('.');
       SingleResourceNameConfig config =
           fullyQualifiedSingleResourceNameConfigsFromProtoFile.get(fullName);
       singleResourceConfigsFromProtoFile.put(fullName.substring(periodIndex + 1), config);
     }
+    // Populate a FixedResourceNameConfigs map, using just the unqualified names.
+    LinkedHashMap<String, FixedResourceNameConfig> fixedResourceConfigsFromProtoFile =
+        new LinkedHashMap<>();
+    for (String fullName : fullyQualifiedFixedResourceNameConfigsFromProtoFile.keySet()) {
+      int periodIndex = fullName.lastIndexOf('.');
+      FixedResourceNameConfig config =
+          fullyQualifiedFixedResourceNameConfigsFromProtoFile.get(fullName);
+      fixedResourceConfigsFromProtoFile.put(fullName.substring(periodIndex + 1), config);
+    }
+
+    List<CollectionConfigProto> allCollectionConfigProtos =
+        new ArrayList<>(configProto.getCollectionsList());
+    configProto
+        .getInterfacesList()
+        .stream()
+        .forEach(i -> allCollectionConfigProtos.addAll(i.getCollectionsList()));
+
+    ImmutableMap<String, SingleResourceNameConfig> singleResourceNameConfigsFromGapicConfig =
+        createSingleResourceNamesFromGapicConfigOnly(
```

Comment:
I did end up making this method, as you had suggested in a prior PR haha
it does keep it cleaner, plus it's reusable in the two situations (parsing proto annotations and not parsing them)